http-server/src/server.rs

```diff
@@ -167,6 +157,56 @@ impl<M> Builder<M> {
 		self
 	}
 
+	/// Finalizes the configuration of the server with customized TCP settings on the socket.
+	///
+	///
+	/// ```rust
+	/// use jsonrpsee_http_server::{HttpServerBuilder, HyperTcpConfig};
+	/// use socket2::{Domain, Socket, Type};
+	/// use std::time::Duration;
+	///
+	/// #[tokio::main]
+	/// async fn main() {
+	///   let addr = "127.0.0.1:0".parse().unwrap();
+	///   let domain = Domain::for_address(addr);
+	///   let socket = Socket::new(domain, Type::STREAM, None).unwrap();
+	///   socket.set_nodelay(true).unwrap();
+	///   socket.set_reuse_address(true).unwrap();
+	///   socket.set_nonblocking(true).unwrap();
+	///   socket.set_keepalive(true).unwrap();
+	///
+	///   let address = addr.into();
+	///   socket.bind(&address).unwrap();
+	///
+	///   socket.listen(4096).unwrap();
+	///
+	///   // hyper does some settings on the provided socket, ensure that nothing breaks our "expected settings".
+	///   let hyper_cfg = HyperTcpConfig { sleep_on_accept_errors: true, keepalive_timeout: Some(Duration::from_secs(1)), no_delay: true };
+	///
+	///   let server = HttpServerBuilder::new().build_from_tcp(socket, hyper_cfg).unwrap();
+	/// }
+	/// ```
+	pub fn build_from_tcp(self, listener: impl Into<StdTcpListener>, cfg: HyperTcpConfig) -> Result<Server<M>, Error> {
```

Comment:
@jsdw is this API reasonable?

Basically the only thing I want to ensure is that no settings gets overwritten without anything informing the users of this API.

It's a bit awkward to some settings twice but no way around that....

Reply:
fair enough, I think your suggestion is less awkward.